src/scos_tekrsa/settings.py

```diff
@@ -13,15 +12,9 @@
 __cmd = Path(sys.argv[0]).name
 
 RUNNING_TESTS = "test" in __cmd
-
-if not settings.configured or not hasattr(settings, "MOCK_SIGAN"):
-    MOCK_SIGAN = env.bool("MOCK_SIGAN", default=False) or RUNNING_TESTS
-else:
-    MOCK_SIGAN = settings.MOCK_SIGAN
-if not settings.configured or not hasattr(settings, "MOCK_SIGAN_RANDOM"):
-    MOCK_SIGAN_RANDOM = env.bool("MOCK_SIGAN_RANDOM", default=False)
-else:
-    MOCK_SIGAN_RANDOM = settings.MOCK_SIGAN_RANDOM
-
-if not settings.configured:
+DEVICE_MODEL = env("DEVICE_MODEL", default="RSA507A")
```

Comment:
The SignalAnalyzerInterface defines a property for [the sigan model](https://github.com/NTIA/scos-actions/blob/b0dd42b0281db1af87d21b162403ef3573bfab06/scos_actions/hardware/sigan_iface.py#L161), and the Tektronix API can provide the device model name. As-is, the `TekRSASigan` creates a `self.device_name` instance variable containing the name/model  of the current device.

Should this environment variable be removed, and the instance variable of the initialized sigan be used instead? I realize there may be a reason to use an environment variable instead based on incoming changes at the SCOS Sensor level.

Reply:
The problem is that scos-sensor will now create the signal analyzer instance and scos-sensor currently simply iterates through the packages that start with "scos_" and calls each package's discover. I think we could add a method to the discover package that would allow us to pass in the signal analyzer, but I'd have to test it out.

Reply:
I made changes in [9e3d509](https://github.com/NTIA/scos-tekrsa/pull/53/commits/9e3d509515a552abd70dff9cf44dd1e5cd82f9b1) to at least use the `model` property from SignalAnalyzerInterface instead of `device_name`, while leaving the discover/settings behavior of using the environment variable unchanged.

I think the environment variable being used for action discovery is OK since modifying the behavior would be a substantial increase in complexity, and other sigan plugins probably would not require the sigan instance to be passed into the discover module.